lib/jekyll/converters/scss.rb

```diff
@@ -75,11 +75,19 @@ def sass_dir_relative_to_site_source
       end
 
       def sass_load_paths
+        paths = (user_sass_load_paths + [sass_dir_relative_to_site_source])
+
         if safe?
-          [sass_dir_relative_to_site_source]
-        else
-          (user_sass_load_paths + [sass_dir_relative_to_site_source]).uniq
-        end.select { |load_path| File.directory?(load_path) }
+          paths.map! { |path| Jekyll.sanitized_path(@config["source"], path) }
+        end
+
+        # Expand file globs (e.g. `node_modules/*/node_modules` )
+        Dir.chdir(@config["source"]) do
+          paths = paths.map { |path| Dir.glob(path) }.flatten.uniq.
+                        map { |path| File.expand_path(path) }
```

Comment:
There's probably some DOS vectors if we don't sanitize before (.e.g. `/**/*` would scan the entire file system). Should we do both?